### PR TITLE
Separate consolidated dispatcher subscription endpoint

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Separate consolidated dispatcher subscription endpoint into its own type
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by lberk** [comment]
/hold
/lgtm
as we chatted, holding to make sure #433 lands smoothly before adding this.

File: pkg/channel/consolidated/dispatcher/subscription_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		return
+	}
+	_, err = w.Write(jsonResult)
+	if err != nil {
+		d.logger.Errorf("Error writing jsonResult to serveHTTP writer: %w", err)
+	}
+}
+
+func (d *subscriptionEndpoint) start() {
+	d.logger.Fatal(nethttp.ListenAndServe(":8081", d))
```
Comment (slinkydeveloper) [inline pkg/channel/consolidated/dispatcher/subscription_endpoint.go:57]: I think this goes fatal if err == nil, so split it with a if err != nil

Comment (aliok) [reply pkg/channel/consolidated/dispatcher/subscription_endpoint.go:59]: From https://golang.org/pkg/net/http/#ListenAndServe

```
ListenAndServe always returns a non-nil error.
```

**Comment by aliok** [comment]
@slinkydeveloper this is rebased.

I guess it is fine if don't ask for others' review since this is only a refactoring w/o any business logic change.

File: pkg/channel/consolidated/dispatcher/subscription_endpoint.go (hunk 1 above)
Comment (aliok) [reply pkg/channel/consolidated/dispatcher/subscription_endpoint.go:59]: :shrug:

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**